test/coverage.ps1

```diff
@@ -0,0 +1,36 @@
+param(
+    [string][Alias('c')]$configuration = "Release",
+    [string][Alias('o')]$output = "CoverageOutput"
+)
+
+function Install-Tools {
+    dotnet tool restore
+}
+
+function Print-Environment {
+    dotnet --version
+    dotnet tool run coverlet --version
+}
+
+function Prepare-Output {
+    if (Test-Path -Path $output) {
+        Remove-Item $output -Recurse
+    }
+}
+
+function Run-Tests {
+    $testProjects = Get-ChildItem -Filter "*.Test.csproj" -Recurse
```

Comment:
I think we'd need to do this in order to create a `watch` script and allow for auto-testing/coverage generation.

Reply:
I totally agree that I would like a bash version of `coverage.ps1` but I will have to admit it is a shortcoming of mine so I will give it my best but don't feel bad to tell me what I am doing wrong. I *think* the `/p:` will work in bash, according to [this](https://stackoverflow.com/questions/51485427/how-to-pass-build-properties-to-dotnet) it only needs to be quoted when you are passing multiple. 

Do you think it would be a good idea to check in the `.vscode/tasks.json` into the repo? I know the repo currently has it in the [.gitignore](https://github.com/bitwarden/server/blob/master/.gitignore#L210) but the "VisualStudio" default now [specifically allows it](https://github.com/github/gitignore/blob/master/VisualStudio.gitignore#L371). 

I am totally down for using a specific `.sln` file for this.  Unless I'm missing something, I'm not sure it would speed up coverage. Running `dotnet test $PSScriptRoot/bitwarden.tests.sln /p:CoverletOutputFormatter="cobertura" --collect:"XPlat Code Coverage" --results-directory:"$output" -c $configuration` would still run all the tests in each `csproj` separately and they would each create their own coverage file that would need to be combined to fully see what's going on. With slight alterations that line could be `watch`ed but it also wouldn't combine the results. 

So I will add a `.sln` file but i'm not sure it fixes some of those points. Should that file be added to the main solutions items or should it be left out because it mostly only exists for this process that has to be invoked from outside of Visual Studio anyways?